vsintegration/src/FSharp.Editor/Hints/InlineParameterNameHints.fs

```diff
@@ -147,21 +147,26 @@ type InlineParameterNameHints(parseResults: FSharpParseFileResults) =
         else
             []
 
-    member _.GetHintsForUnionCase (symbol: FSharpUnionCase) (symbolUse: FSharpSymbolUse) =
+    member _.GetHintsForUnionCase (sourceText: SourceText) (symbol: FSharpUnionCase) (symbolUse: FSharpSymbolUse) =
         if isUnionCaseValidForHint symbol symbolUse then
 
             let fields = Seq.toList symbol.Fields
 
             let ranges =
                 parseResults.GetAllArgumentsForFunctionApplicationAtPosition symbolUse.Range.Start
 
+            let argumentNames =
+                match ranges with
+                | Some ranges -> (List.map (getSourceTextAtRange sourceText)) ranges
+                | None -> []
             // When not all field values are provided (as the user is typing), don't show anything yet
             match ranges with
             | Some ranges when ranges.Length = fields.Length ->
                 fields
                 |> List.zip ranges
-                |> List.where (snd >> fieldNameExists)
-                |> List.map getFieldHint
+                |> List.zip argumentNames
```

Comment:
maybe try `List.zip3` here? then the matches below would be `(fun (argumentName, range, parameter) -> ` and would only take one 'iteration'.

In addition, a `where` + `map` is a `choose`, so those could be combined for efficiency.

Reply:
Thanks for the feedback. About using `choose` wouldn't that return an option list? \